### ISSUE TITLE
Future call / cron capability for serverless mode

### ISSUE THREAD

**Issue by lukehutch** [serverpod/serverpod#2168]
I need to be able to run a process once a day to collect daily user stats. I was planning on using future calls for this, but then I just saw this in the docs:

> The future call feature is not enabled when running Serverpod in serverless mode.

Can you please create an easy way to add cron tasks to the Postgres Docker container, including deploying Dart code to the container that depends upon the server project?

**Comment by Isakdl** [comment]
In a production environment, you can setup a cloud cron job such as [cloud scheduler in GCP](https://cloud.google.com/scheduler/) and then run Serverpod in maintenance mode `dart run bin/main.dart --role maintenance`. The server will boot complete any pending future calls and then shut down.

**Comment by lukehutch** [comment]
OK, so if I want a job to run every day, I need to have the `main.dart` actually insert a future call every time it is called, so that it has a future to complete? I don't think there are recurring calls right now, like `Timer.periodic`...

This would be a lot easier if the `ServersideClient` existed, which I proposed in #1009 -- then I could just call the server from a "client" app that could be run as a cron job.

**Comment by Isakdl** [comment]
Yes correct! Indeed that would be pretty neat solution. 

Well you could implement a very small main.dart that has access to the generated client that simply just calls the endpoint. Then run that main.dart as a cron job in a similar service as I linked above and you can also configure retries on error etc.

I think that workaround would be pretty robust for your use-case, until we have first-class support.

**Comment by lukehutch** [comment]
Yes, that's what I plan to do. However I don't have the authentication part of that figured out yet. Iighr have to spoof email/password-based login. But I don't want to have to manually create an admin account when I deploy the system. I'd rather have some sort of admin authentication handler that was built in, and accepted a specific password, without the extra email authentication machinery.
